README.md
## Air-light - A minimalist WordPress starter theme

[![Build Status](https://img.shields.io/travis/com/digitoimistodude/ak-starter.svg?style=flat-square)](https://travis-ci.com/digitoimistodude/ak-starter) [![GitHub release](https://img.shields.io/github/tag/digitoimistodude/ak-starter.svg?style=flat-square)](https://github.com/digitoimistodude/ak-starter/releases) ![GitHub contributors](https://img.shields.io/github/contributors/digitoimistodude/ak-starter.svg?style=flat-square)

Air-light (or simply *Air*) is designed to be a ultra minimal starting point for a WordPress project at [Digitoimisto Dude Oy](https://www.dude.fi), a Finnish boutique digital agency in the center of Jyväskylä. Theme is originally based on [\_s](https://github.com/automattic/_s). We welcome all happy contributors with open arms! [See roadmap](https://favro.com/organization/3b45e73eaf083f68fefef368/c1dd2d4a99d6723904d2e763).

## [Demo](https://airwptheme.com/demo) | [Documentation](https://github.com/digitoimistodude/ak-starter/wiki)

* **CSS gzipped:** 16.8 KB *(originally 124.4 KB)*
* **JS gzipped:** 8.6 KB *(28.6 KB original)*
* **Front page HTML**: 7.4 KB *(29.4 KB original)*

![ak-starter-readme-mockup-latest](https://user-images.githubusercontent.com/1534150/170326942-b7404051-7621-4df3-9294-0ccd5e6f59a7.png)

## Weekly updates!

This theme is constantly kept up to date by a bunch of [awesome contributors](https://github.com/digitoimistodude/ak-starter/graphs/contributors). Wanna join in development? Read [the instructions for contributing](#contributing) and let us know about your first PR!

![Alt](https://repobeats.axiom.co/api/embed/bd95f5747b41d3cf662e033044af08dee9a15f9f.svg "Repobeats analytics image")

### Mission & vision

Air-light is built to be very straightforward, backwards compatible, front-end developer friendly and modular by its structure. Following [Underscores](https://github.com/automattic/_s) and [WordPress Theme Coding Standards](https://codex.wordpress.org/Theme_Development#Theme_Development_Standards) best practices and most of the changes in \_s are implemented as soon as they are committed.

**Our mission and goal** is **minimalism** and **simplicity**. **Our vision** is to build a theme that will not implement its own wrappers or functions, will not use any templating languages that would take things further from traditional PHP or CSS, will contain nothing that people will not use or need. Air-light will be free of weird "app-like" folder structures or odd syntaxes that nobody else uses. We love WordPress as it was and as it is.

Air was renamed to ak-starter in version 3.7.8 (March 20th, 2018), because *air* was already taken in the official WordPress theme directory.

### Official, approved, accessibility-ready!

Air-light v. 4.2.2 was approved to [official WordPress theme directory](https://wordpress.org/themes/ak-starter/) on June 4, 2018. But please note, all changes you do to the theme without generating your own or changing textdomain will be overridden in theme updates - so if you use this theme as a starting point, please follow instructions and/or replace the textdomain with your own.

![accessibility](https://user-images.githubusercontent.com/1534150/155683656-aab335d3-8e11-4600-ade7-f7c332ade87d.gif)
_Fully keyboard accessible navigation!_

## Table of contents

1. [Usage](#usage)
2. [Please note before using](#please-note-before-using)
3. [License](#license)
4. [Features](#features)
    1. [Layout base & grid](#layout-base--grid)
    2. [Typography](#typography)
    3. [Development](#development)
    4. [Navigation](#navigation)
    5. [WordPress & functions](#wordpress--functions)
    6. [Custom Post Types](#custom-post-types)
    7. [Custom Taxonomies](#custom-taxonomies)
    8. [Namespaced PHP](#namespaced-php)
    9. [Accessibility](#accessibility)
    10. [Lazy load](#lazy-load)
    11. [Disabled features](#disabled-features)
5. [Extra building blocks](#extra-building-blocks)
    1. [Sticky navigation](#sticky-navigation)
    2. [WooCommerce support](#woocommerce-support)
6. [Requirements](#requirements)
7. [Recommendations for development](#recommendations-for-development)
8. [How to build a new theme](#how-to-build-a-new-theme)
9. [Contributing](#contributing)
    1. [Air development](#air-development)
    2. [Debuggers](#debuggers)
        1. [For Gulp](#for-gulp)
        2. [For your editor](#for-your-editor)
    3. [Releasing a new version (staff only)](#releasing-a-new-version-staff-only)
10. [Notes](#notes)

### Please note before using

Air is a **development theme**, so it has updates very often. By using this starter theme, you agree that the anything can change to a different direction without a warning when you look at this dev-git the next time. Please note this theme has no updates inside WordPress by design. Use this theme to hack it to pieces and create your new awesome theme based on Air! Please also see [Debuggers](#debuggers)!

Air is not meant to be "a theme for everyone", which means it doesn't have all the parts that are generally included in multi-purpose themes for non-technical people (please see [Disabled features](#disabled-features)).

If you want to use this theme as starter for your new theme, please note the theme won't necessarily be that much fun or won't look good by default and needs work from you. We recommend using [Sage](https://roots.io/sage/) if you need something more extended.

### License

Air is licensed with [The MIT License (MIT)](http://choosealicense.com/licenses/mit/) which means you can freely use this theme commercially or privately, modify it, or distribute it, but you are forbidden to hold Dude liable for anything, or claim that what you do with this is made by us.

## Theme structure

We try to follow traditional WordPress [Template Hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy) as much as possible. This way we will provide a low threshold for junior developers and go hand in hand with official [WordPress Theme Coding Standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/).

Please see [Visual Overview](https://developer.wordpress.org/themes/basics/template-hierarchy/#visual-overview) if you are interested in how the WordPress theme structure works.

Here's the current Air-light theme structure (prone to changes, not updating in real time, but stays mostly the same in its pricinpals):

``` shell
themes/your-theme-name/              # → Root of your ak-starter based theme
├── 404.php                          # → Default "not found" page
├── archive.php                      # → Default archive template
├── bin/                             # → Scripts
│   ├── ...                          # → This scripts are used for generating themes or new Air-light version
│   └── newtheme.sh                  # → The start script for creating YOUR own theme out of Air-light
├── comments.php                     # → Default comments template (can be deleted if not needed)
├── css/                             # → CSS files for production (never edit)
│   ├── dev/                         # → Unminified stylesheet files for debugging (never edit)
│   └── prod/                        # → Minified stylesheet files for production (never edit)
├── fonts/                           # → Your webfont files (woff, woff2, ttf needed)
├── footer.php                       # → Site footer
├── front-page.php                   # → Demo front-page template (not included in wordpress.org version)
├── functions.php                    # → Set up your theme basic settings
├── gulp/                            # → Gulp related settings and tasks
│   └── ...                          #
├── gulpfile.js                      # → Core gulpfile for ak-starter development
├── header.php                       # → Site header
├── images/                          # → Your theme images, for example default featured images and placeholders
├── inc/                             # → Theme core PHP
│   ├── hooks/                       # → Hook functions
│   ├── includes/                    # → Non-template features
│   ├── template-tags/               # → Template functions and helpers
│   ├── post-types/                  # → Custom Post Types
│   ├── taxonomies/                  # → Custom Taxonomies
│   ├── hooks.php                    # → All hooks the theme runs are here
│   ├── includes.php                 # → Include non-template features
│   └── template-tags.php            # → Include template functions and helpers
├── js/                              # → JavaScript files
│   ├── dev/                         # → Unminified script files for debugging (never edit)
│   ├── prod/                        # → Minified script files for production (never edit)
│   └── src/                         # → Script files for development (edit these)
│       ├── navigation.js            # → Accessible multi-level navigation (from 3.4.5)
│       ├── front-end.js             # → Theme core JavaScript file (from 1.0.0, before: scripts.js)
│       └── sticky-nav.js            # → Sticky nav functionality (optional)
├── package.json                     # → Node.js dependencies and scripts
├── page.php                         # → Default page template
├── phpcs.xml                        # → PHPCodeSniffer/WordPress Theme Coding Standards settings
├── sass/                            # → SCSS files for CSS development
│   ├── base/                        # → Theme base styles
│   │   └── _accessibility.scss      # → Accessibility
│   ├── global.scss                  # → Core CSS file that calls all the modular files
│   ├── gutenberg-editor-styles.scss # → Core CSS file for Gutenberg editor and blocks
│   ├── components/                  # → Add your style components to this folder
│   ├── features/                    # → Fuctionality styles
│   │   ├── _gallery.scss            # → Default WordPress gallery feature styles
│   │   ├── _gravity-forms.scss      # → Defaults for Gravity Forms + WCAG 2.0 form fields for Gravity Forms
│   │   ├── _lazyload.scss           # → Styles for air-helper lazyload feature (lazyload.js needed)
│   │   ├── _top.scss                # → Back to top styles
│   │   ├── _pagination.scss         # → Numbered pagination styles
│   │   └── _sticky-nav.scss         # → Sticky nav styles (not included by default)
│   ├── gutenberg/                   # → Gutenberg block styles for both core and custom blocks
│   │   └── ...                      # → WIP
│   ├── helpers/                     # → Helper mixins and functions
│   │   ├── _animations.scss         # → Animations and effects
│   │   ├── _aspect-ratio.scss       # → A mixin for aspect ratio
│   │   ├── _general.scss            # → Mixins for general use, or helpers of other mixins
│   │   ├── _grid.scss               # → CSS Grid helper mixin
│   │   └── _typography.scss         # → Typography style mixins
│   ├── layout/                      # → Fuctionality styles
│   │   ├── _forms.scss              # → Styles for general forms and Gravity Forms
│   │   ├── _site-footer.scss        # → Footer styles
│   │   ├── _site-header.scss        # → Header styles
│   │   ├── _typography.scss         # → Defaults for typography and fonts
│   │   └── _gutenberg.scss          # → Site-side styles for Gutenberg (pratically for single.php)
│   ├── navigation/                  # → Navigation styles
│   │   ├── _burger.scss             # → Burger styles and animations
│   │   ├── _nav-desktop.scss        # → Desktop navigation styles and dropdowns
│   │   └── _nav-mobile.scss         # → Navigation styles for mobile and touch devices
│   ├── variables/                   # → Configurations
│   │   ├── _breakpoints.scss        # → Widths from mobile to TV screens
│   │   ├── _colors.scss             # → All the colors of the theme
│   │   ├── _font-face.scss          # → Define font variables here
│   │   ├── _font-family.scss        # → Define font paths
│   │   ├── _font-size.scss          # → Define font related sizes
│   │   ├── _forms.scss              # → Form related variables
│   │   └── _spacings.scss           # → Margins and paddings
│   ├── views/                       # → Templates, archives, pages and views go here
│   │   ├── _404.scss                # → Default 404 styles
│   │   ├── _comments.scss           # → Comment styles (optional)
│   │   ├── _front-page.scss         # → Front page styles (empty by default)
│   │   └── _search.scss             # → Default search result styles
├── screenshot.png                   # → Theme screenshot for WP admin
├── search.php                       # → Default search view
├── sidebar.php                      # → Default sidebar (optional)
├── single.php                       # → Default single article or CPT view
├── style.css                        # → Theme meta information
├── svg/                             # → Your theme SVG graphics and icons
└── template-parts/                  # → WordPress template parts. Modules go under this folder.
    ├── header/                      # → Header modules
    │   ├── branding.php             # → Site branding
    │   ├── navigation.php           # → Site navigation
    └── hero.php                     # → Default hero
```

### Features

Some features, WooCommerce support and personal preferences of Dude are moved to [Air helper](https://github.com/digitoimistodude/air-helper) plugin.

#### Layout base & grid

* Gutenberg-ready
* Flexbox-ready
* CSS Grid-ready
* SVG-ready
* [SASS](http://sass-lang.com/)-support (SCSS-syntax)
* CSS reset from [sanitize.css](https://github.com/csstools/sanitize.css)
* Section blocks and containers
* Basic and minimal CSS framework for forms, commenting and typography

#### Typography

* Responsive typography with viewport units with fallbacks, see more in [sass/layout/\_typography.scss](https://github.com/digitoimistodude/air/blob/master/sass/layout/_typography.scss) and [sass/base/\_helpers.scss](https://github.com/digitoimistodude/air/blob/master/sass/base/_helpers.scss#L91), default syntax is `@include responsive-font($font-size-min, $font-size-max);` (formerly [Megatype](https://github.com/StudioThick/megatype), still recommended with blogs or text-only based sites, but **not** included by default after 1.5.0)
* Web fonts file are preferred, helper included: [Sass Boilerplate's fontFace-mixin](https://github.com/magnetikonline/sassboilerplate/blob/master/fontface.scss). Put files in `fonts/` directory, you'll need .odt, .ttf, .woff, .woff2. Then just `@include fontFace('Proxima Nova', '../../fonts/proximanova-regular-webfont', 400);` in \_typography.scss.

#### Development

* [BrowserSync](http://www.browsersync.io/) for keeping multiple browsers and devices synchronized while testing, along with injecting updated CSS and JS into your browser while you're developing (included in [devpackages](https://github.com/digitoimistodude/devpackages))
* [gulp](http://gulpjs.com/) build script that compiles both Less and Sass, checks for JavaScript errors, optimizes images, and concatenates and minifies files (see Dude's [devpackages](https://github.com/digitoimistodude/devpackages))
* [npm](https://www.npmjs.com) for front-end package management

#### Navigation

* WCAG 2.0 accessible with keyboard and screen readeres, aria roles and labels included
* Custom navigation walker
* Support for multi-level drop down submenus
* Support for both absolute and relative navigation
* Improved version of the [accessible menu for WordPress themes](https://github.com/theme-smith/accessible-nav-wp), fully accessible and responsive multi-level navigation
* Support for animations
* Hover intent with minimal animations
* Accessible mobile menu
* Wide selection of [tasty hamburger](https://github.com/jonsuh/hamburgers) animations

#### WordPress & functions

* Available for translation, full Polylang Pro support
* Support for Post Thumbnails on posts and pages
* Gutenberg support
* HTML5 core markup for WordPress elements
* **Air specific:** Hero template, section *blocks*

#### Custom Post Types

Air-light can register your CPT:s automatically.

1. Add your custom post type to theme settings under post\_types, located in `functions.php` like this:

```
'post_types' => [
  'your-post-type' => 'Your_Post_Type'
]
```

2. Add a file `inc/post-types/your-post-type.php`
3. Extend `Post_Type` class with `Your_Post_Type` and define your post type in a public function called `register()`. See the example: `inc/post-types/your-post-type.php`.

#### Custom Taxonomies

Air-light can register your Taxonomies automatically.

1. Add your taxonomy to theme settings under taxonomies, located in `functions.php` like this:

```
'your-taxonomy' => [
  'name' => 'Your_Taxonomy'
  'post_types' => 'post, page'
]
```

2. Add a file `inc/taxonomies/your-taxonomy.php`
3. Extend `Taxonomy` class with `Your_Taxonomy` and define your taxonomy in a public function called `register()`. See the example: `inc/taxonomies/your-taxonomy.php`.

### Namespaced PHP

Air-light uses namespaced PHP since 5.0.0. This means that we no longer need to prefix functions and hooks, because `namespace Air_Light;` takes care of that.

When old function format was:

``` php
// Pre_get_posts
add_action( 'pre_get_posts', 'dude_pre_get_posts' );
function dude_pre_get_posts( $query ) {
  // Do something
}
```

New format goes like this:

``` php
// Pre_get_posts
add_action( 'pre_get_posts', __NAMESPACE__ . '\pre_get_posts' );
function pre_get_posts( $query ) {
  // Do something
}
```

#### Accessibility

Creating accessible websites is really important and our goal is to make air as accessible-ready as possible. Theme fully supports navigating with keyboard and screen-readers. Other accessible features:

* [Navigation patterns](#navigation)
* Skip to content link
* Smart focus for keyboard users, [what-input](https://github.com/ten1seven/what-input) baked in
* Valid HTML
* Accessible SVG icons
* Screen reader class
* External link indicators
* Underlined links
* Content-aware back to top link
* WCAG 2.0 AAA Accessible-ready Gravity Forms styles (needs [WCAG 2.0 form fields for Gravity Forms](https://wordpress.org/plugins/gravity-forms-wcag-20-form-fields/), included in [dudestack](https://github.com/digitoimistodude/dudestack))

#### Lazy load

From 4.7.1 ak-starter has a lazy loading image features for background images and imgs. If you don't use this feature, remove it from [scripts](https://github.com/digitoimistodude/blob/d517688bb400ac68100dcbdcd4bc7dbce7739099/js/src/scripts.js#L10). This feature depends on [air-helper](https://github.com/digitoimistodude/air-helper), check out [the documentation in air-helper](https://github.com/digitoimistodude/air-helper#image-lazyloading-1) for further instructions.

#### Disabled features

* Widgets
* Post formats
* Jetpack support
* (Threaded) comments
* Underscores Template tags
* Sidebar

### Javascript

#### Structure

All .js files in `/js/src/*` is built to production bundles in `/js/prod/` folder and development bundles in `/js/dev/` folder with the same name and loaded corresponding to the WP_ENV environment variable. The production scripts can be loaded on development by using `?load_production_builds` URL parameter. The main scripts file loaded in front end is `/js/src/front-end.js`.

If you want to add a piece of custom JS, create a file under `/js/src/modules/` and import or require it in `/js/src/front-end.js`. If you need a admin-specific JS, add a `/js/src/admin.js` and then enqueue `/js/dist/admin.js` with `enqueue_admin_scripts`

#### Legacy support

Our build uses babel to translate scripts to ES2015 compatible JS, so you can use modern JS syntax without thinking about backwards compatibility. There is a `/js/src/legacy.js` file, which contains the needed polyfills for browsers not supporting the ES2015 syntax and is automatically loaded on the header when such browser is detected.

#### Linter

We use [Airbnb](https://github.com/airbnb/javascript) es-lint presets spiced up with our own flavors.

### Extra building blocks

#### Sticky navigation

Air has a sticky navigation baked in.

##### How to enable

You can enable the navigation by

1. Adding sticky-nav.js to your gulpfile (already included with [Devpackages](https://github.com/digitoimistodude/devpackages) and bin/newtheme.sh start script)
2. Uncommeting sticky-nav import in global.scss
3. Restart gulp and save scripts.js once to compile working combined javascript file

#### WooCommerce support

Air had by default a basic WooCommerce support [from version 1.9.2](https://github.com/digitoimistodude/commit/55c539bb9cd2e35fdbfdf4f39a136c542b42b884), and for a while it was been separated to its own repository, [air-woocommerce](https://github.com/digitoimistodude/air-woocommerce) since v2.5.6.

##### How to enable

Starting from v2.6.0 WooCommerce support comes with [Air helper](https://github.com/digitoimistodude/air-helper) plugin and Air contains optional very basic WC styles. Air helper will add it's WC functionality when theme support for WooCommerce is added. To enable:

1. Get [Air helper](https://github.com/digitoimistodude/air-helper)
2. Activate the plugin

### Requirements

* PHP >= 7.4
* Requires at least: WordPress 4.7.1
* Tested up to WordPress 6.0.0

### Recommendations for development

* macOS
* [Devpackages](https://github.com/digitoimistodude/devpackages) \- Npm and Gulp \+ plugins
* [Dudestack](https://github.com/digitoimistodude/dudestack) \- A toolkit for creating a new professional WordPress project with deployments\. Heavily based on Bedrock by Roots\.

### How to build a new theme

Please refer to Wiki section [Getting started in theming with Air-light](https://github.com/digitoimistodude/ak-starter/wiki#2-getting-started-in-theming-with-ak-starter).

### Usage

See [Documentation](https://github.com/digitoimistodude/ak-starter/wiki).

### Contributing

If you have ideas about the theme or spot an issue, please let us know. Before contributing ideas or reporting an issue about "missing" features or things regarding to the nature of that matter, please read [Please note](#please-note-before-using) section. Thank you very much.

If you want to contribute to the development, please follow these instructions:

1. Create a fork of this repository (or new branch if you have editor/maintainer permissions)
2. Make your changes
3. Create a pull request

### Air development

#### Installation

If you want to improve air, you have two options.

##### 1\. Use dudestack

Air is originally built on [dudestack](https://github.com/digitoimistodude/dudestack). Install our development environment with these steps (unix only, sorry Windows!):

1. `mkdir -p /var/www && cd /var/www/ && git clone https://github.com/digitoimistodude/dudestack`
2. Go to bin folder `cd /var/www/dudestack/bin` and run `bash wsl.sh` or `bash macos.sh` depending on your platform. Follow instructions.
3. Run `createproject`, name project after *airdev* when asked
4. Wait for the project to be created (get a coffee, first time can take couple of minutes)
5. Create a fork of ak-starter (press fork button on GitHub) (or if you are Dude staff, just create new branch for changes)
6. Go to the theme folder of your WordPress instance via Terminal (`cd /var/www/airdev/content/themes`)
7. Clone your fork with `git clone git@github.com:yourusername/ak-starter.git` (replace yourusername with your actual username)
8. Cd to your new cloned repository `cd /var/www/airdev/content/themes/ak-starter`
9. Get the dependencides by running `npm install` inside the theme folder (if you don't have npm installed, see [here](https://www.npmjs.com/get-npm) or just use [homebrew](https://brew.sh))
10. Wait npm to get through files (get another cup of coffee)
11. Activate theme - if you are using the lightweight [macos-lemp-setup](https://github.com/digitoimistodude/macos-lemp-setup): `cd /var/www/airdev && vendor/wp-cli/wp-cli/bin/wp theme activate ak-starter`
12. Open whole project to your preferred coding editor, for example when using [Visual Studio Code](https://github.com/ronilaukkarinen/vscode-settings) that would be `code /var/www/airdev/content/themes/ak-starter` or via GUI (Open folder).

CHANGELOG.md
### [Unreleased]

* Add box-model helper

### 9.0.6: 2022-06-30

* Upgrade to [devpackages 2.4.7](https://github.com/digitoimistodude/devpackages/releases/tag/2.4.7)
* Remove air_helper_custom_settings_post_ids hook that is no longer used (kudos to latenssi @ wpfi Slack for spotting it!)

### 9.0.5: 2022-06-09

* Prevent mobile navigation flickering before JS has been loaded

### 9.0.4: 2022-06-09

* Add default localization for accessible carousels to support Air-blocks 1.1.2

### 9.0.3: 2022-06-08

* Remove leftover %screen-reader-text from editor styles
* Move editor styles outside of the global gutenberg-editor-styles.scss
* Fix form elements in ACF block previews
* A11y: Localizations for possible carousel blocks from Air-blocks

### 9.0.2: 2022-05-27

* Fix critical regression about ACF blocks not outputting, Fixes #153

### 9.0.1: 2022-05-26

* Disable dropdown box-shadow by default
* Improve placeholder for new project
* Fix typo in CSS variable name
* Fix critical JS error with calling getChildAltText() in vain
* Fix outdated release command

### 9.0.0: 2022-05-24

#### Major update

* Rebranding: Update default logo
* Deprecate %screen-reader-text, use mixin instead
* Deprecate old lazy load styles from future themes
* Add var(--box-shadow-sub-menu) for desktop navigation
* Add more clear starting point with demo content placeholder
* Add support for anchors in the block (thanks @timiwahalahti)
* Disable support for additional CSS classes in blocks (thanks @timiwahalahti)
* Bump tested up WordPress version to 6.0
* Fix back to top label for screen readers
* Add data-version attribute to footer indicating Air-light version in use

### 8.4.7: 2022-05-24

* Remove _slick.scss, deprecate slick-carousel slider (carousel block is part of [air-blocks](https://github.com/digitoimistodude/air-blocks) now)

### 8.4.6: 2022-05-23

* Combine media queries
* Deprecate heading-hero component

### 8.4.5: 2022-05-20

* Use only one burger animation, revert burger style on mobile

### 8.4.4: 2022-05-20

* Simplify external link selector
* Add js-trigger to skip-link
* Combine nav-mobile media queries
* Update stylelintrc rules block-closing-brace-newline-after and at-rule-empty-line-before
* Fix cubic-bezier mixin $duration variable
* Fix iOS 15 nav width style bug
* Fix iOS 15 scrolling bug with `height: -webkit-fill-available;`
* New navigation animation: slide (default), change to fade by setting `$nav-slide-animation: false;`

### 8.4.3: 2022-05-13

* Fixes #134, nav-toggle menu position shifting on iOS 15.1

### 8.4.2: 2022-05-12

* Add anchors as JS module
* Deprecate vanilla-lazyload and support for older JS-based lazy loading methods
* Drop jQuery
* Rewrite accessible back to top functionality and styles
* Combine external link labels with external link JS module
* Refactor front-end.js

### 8.4.1: 2022-05-12

* Fix paddings on editor
* Remove deprecated text columns block
* Remove default list-type from typography scss and use core-list instead
* Fix responsive lists on gutenberg editor

### 8.4.0: 2022-05-12

* Fix cover block default text color
* Fix overflow in stacked blocks
* Fix core-list not working properly on Gutenberg editor preview
* Fix navigation dropdown menus opening inside viewport after resizing window
* Fix code block style bugs
* Fix pullquote blockquote alignment
* Fix naming of core-button to core-buttons
* Use var(--color-blockquote) color variable in pull quotes
* Simplify core-list block
* Smaller external link icon
* Remove deprecated core-gallery styles, use Gutenberg core styles instead
* Scroll to top when triggering mobile navigation to ensure no gaps are between header and navigation (thanks @Tumppex!)
* Simplify link formatting

### 8.3.8: 2022-04-29

* Fix variable name in link formatting (--color-link-text-hover)
* Add proxyUrl (devpackages 2.4.5)
* Fix config.styles.src that causes gutenberg styles not to compile (devpackages 2.4.6)

### 8.3.7: 2022-04-11

* Remove leftover code from Gutenberg editor
* Move externalLinkDomains list to theme settings for consistency (devpackages 2.4.4)
* Add cssnano and related postcss-plugins, deprecate gulp-clean-css that is in maintenance-mode (devpackages 2.4.4)
* Improve watch task to be more performant (devpackages 2.4.4)
* Add gulp-size and verbose information to console (devpackages 2.4.4)
* Add instructions on how to contribute (devpackages 2.4.4)
* Update stylelint-config-standard, stylelint-config-recommended-css, caniuse-lite and js related packages (devpackages 2.4.4)
* Get phpcs task from devpackages 2.4.4
* Bump tested up WordPress version to 5.9.3

### 8.3.6: 2022-03-31

* Update browser-sync to 2.27.9 (devpackages 2.4.3)
* Fix main title width in article editor
* Add fonts and font-smoothing in article view of the Gutenberg editor

### 8.3.5: 2022-03-25

* Fix: Margin reset breaks article blocks in editor, limit it for ACF blocks only

### 8.3.4: 2022-03-16

* Make edit-post-visual-editor__post-title-wrapper background to match WordPress brand color
* Bump tested up WordPress version to 5.9.2

### 8.3.3: 2022-03-08

* Upgrade to [devpackages 2.4.2](https://github.com/digitoimistodude/devpackages/releases/tag/2.4.2) (details below)
* Upgrade stylelint to 14.5.3
* Update .stylelintrc rules as per the [official recommendations](https://github.com/stylelint/stylelint-config-recommended/issues/157#issuecomment-1056967465)
* Use kebab-kase in fontface mixin from now on
* Bump tested up WordPress version to 5.9.1
* Upgrade eslint to 8.10.0
* Upgrade eslint-config-airbnb to 19.0.4
* Remove deprecated babel-eslint and use @babel/eslint-parser instead
* Revove outdated and unmaintained gulp-eslint and use gulp-eslint-new instead
* Allow js/src/front-end.js to be linted, fix file for JS warnings

### 8.3.2: 2022-02-10

* Reset blockquote margins as it breaks the auto layout in some situations
* Deprecate hiddentext() mixin that is not used anymore

### 8.3.1: 2022-02-08

#### Major update


functions.php
<?php
/**
 * Gather all bits and pieces together.
 * If you end up having multiple post types, taxonomies,
 * hooks and functions - please split those to their
 * own files under /inc and just require here.
 *
 * @Date: 2019-10-15 12:30:02
 * @Last Modified by:   Roni Laukkarinen
 * @Last Modified time: 2022-06-30 16:36:48
 *
 * @package ak-starter
 */

namespace Air_Light;

/**
 * The current version of the theme.
 */
define( 'AIR_LIGHT_VERSION', '9.0.6' );

// We need to have some defaults as comments or empties so let's allow this:
// phpcs:disable Squiz.Commenting.InlineComment.SpacingBefore, WordPress.Arrays.ArrayDeclarationSpacing.SpaceInEmptyArray

/**
 * Theme settings
 */
add_action( 'after_setup_theme', function() {
  $theme_settings = [
    /**
     * Theme textdomain
     */
    'textdomain' => 'ak-starter',

    /**
     * Image and content sizes
     */
    'image_sizes' => [
      'small'   => 300,
      'medium'  => 700,
      'large'   => 1200,
    ],
    'content_width' => 800,

    /**
     * Logo and featured image
     */
    'default_featured_image'  => null,
    'logo'                    => '/svg/logo.svg',

    /**
     * Custom setting group settings when using Air setting groups plugin.
     * On multilingual sites using Polylang, translations are handled automatically.
     */
    'custom_settings' => [
      // 'your-custom-setting' => [
      //   'id' => Your custom setting post id,
      //   'title' => 'Your custom setting',
      //   'block-editor' => true,
      //  ],
    ],

    'social_media_accounts'  => [
      // 'twitter' => [
      //   'title' => 'Twitter',
      //   'url'   => 'https://twitter.com/digitoimistodude',
      // ],
    ],

    /**
     * All links are cheked with JS, if those direct to external site and if,
     * indicator of that is included. Exclude domains from that check in this array.
     */
    'external_link_domains_exclude' => [
      'localhost:3000',
      'airdev.test',
      'airwptheme.com',
    ],

    /**
     * Menu locations
     */
    'menu_locations' => [
      'primary' => __( 'Primary Menu', 'ak-starter' ),
    ],

    /**
     * Taxonomies
     *
     * See the instructions:
     * https://github.com/digitoimistodude/ak-starter#custom-taxonomies
     */
    'taxonomies' => [
      // 'your-taxonomy' => [
      //   'name' => 'Your_Taxonomy',
      //   'post_types' => [ 'post', 'page' ],
      // ],
    ],

    /**
     * Post types
     *
     * See the instructions:
     * https://github.com/digitoimistodude/ak-starter#custom-post-types
     */
    'post_types' => [
      // 'your-post-type' => 'Your_Post_Type',
    ],

    /**
     * Gutenberg -related settings
     */
    // Register custom ACF Blocks
    'acf_blocks' => [
      // [
      //   'name'           => 'block-file-slug',
      //   'title'          => 'Block Visible Name',
      //   // You can safely remove lines below if you find no use for them
      //   'prevent_cache'  => false, // Defaults to false,
      //   // Icon defaults to svg file inside svg/block-icons named after the block name,
      //   // eg. svg/block-icons/block-file-slug.svg
      //   //
      //   // Icon setting defines the dashicon equivalent: https://developer.wordpress.org/resource/dashicons/#block-default
      //   // 'icon'  => 'block-default',
      // ],
    ],

    // Custom ACF block default settings
    'acf_block_defaults' => [
      'category'          => 'ak-starter',
      'mode'              => 'auto',
      'align'             => 'full',
      'post_types'        => [
        'page',
      ],
      'supports'  => [
        'align'           => false,
        'anchor'          => true,
        'customClassName' => false,
      ],
      'render_callback'   => __NAMESPACE__ . '\render_acf_block',
    ],

    // Restrict to only selected blocks
    // Set the value to 'all' to allow all blocks everywhere
   'allowed_blocks' => [
      'default' => [
      ],
      'post' => [
        'core/archives',
        'core/audio',
        'core/buttons',
        'core/categories',
        'core/code',
        'core/column',
        'core/columns',
        'core/coverImage',
        'core/embed',
        'core/file',
        'core/freeform',
        'core/gallery',
        'core/heading',
        'core/html',
        'core/image',
        'core/latestComments',
        'core/latestPosts',
        'core/list',
        'core/more',
        'core/nextpage',
        'core/paragraph',
        'core/preformatted',
        'core/pullquote',
        'core/quote',
        'core/block',
        'core/separator',
        'core/shortcode',
        'core/spacer',
        'core/subhead',
        'core/table',
        'core/textColumns',
        'core/verse',
        'core/video',
      ],
    ],

    // If you want to use classic editor somewhere, define it here
    'use_classic_editor' => [],

    // Add your own settings and use them wherever you need, for example THEME_SETTINGS['my_custom_setting']
    'my_custom_setting' => true,
  ];

  $theme_settings = apply_filters( 'air_light_theme_settings', $theme_settings );

  define( 'THEME_SETTINGS', $theme_settings );
} ); // end action after_setup_theme

/**
 * Required files
 */
require get_theme_file_path( '/inc/hooks.php' );
require get_theme_file_path( '/inc/includes.php' );
require get_theme_file_path( '/inc/template-tags.php' );

// Run theme setup
add_action( 'init', __NAMESPACE__ . '\theme_setup' );
add_action( 'after_setup_theme', __NAMESPACE__ . '\build_theme_support' );
